### DISCUSSION TITLE
Is "controlled" mode not recommended?

### DISCUSSION THREAD

**Discussion by ZeCarlosCoutinho** [gilbarbara/react-joyride#761] [Q&A] [answered]
In the docs, the prop `stepIndex` has the following message below:
![image](https://user-images.githubusercontent.com/15149259/145031675-4b6779f9-649d-4bf5-9317-d6a8857c2cf8.png)

Is this recommendation here because it's easier to use the uncontrolled version or because some things might break/be suboptimal when using the controlled version?

**Comment by gilbarbara** [comment, marked as answer]
hey @ZeCarlosCoutinho 

No, there's nothing against the controlled mode.
However, using it lifts the tour state to be the app's responsibility which can be tricky...

**Reply by ZeCarlosCoutinho** [reply]
Oh, wow! Such a quick response! Thank you!

From what I understand, the only thing that needs to be controlled by the parent is the `stepIndex` and how it is updated using the callbacks, right?
So, is it tricky because understanding the events/actions/lifecycle is complicated, or is there more to it?

**Reply by gilbarbara** [reply]
That's it
